task-native/src/main/java/org/ballerinalang/stdlib/task/actions/TaskActions.java

```diff
@@ -42,94 +38,92 @@
+        } catch (SchedulerException e) {
+            return Utils.createTaskError(e.getMessage());
         }
         return null;
     }
 
     public static Object resume(BObject taskListener) {
-        Task task = (Task) taskListener.getNativeData(NATIVE_DATA_TASK_OBJECT);
+        String triggerID = (String) taskListener.getNativeData(TaskConstants.TRIGGER_NAME);
+        TaskScheduler taskScheduler = (TaskScheduler) taskListener.getNativeData(TaskConstants.SCHEDULER);
         try {
-            task.resume();
-        } catch (SchedulingException e) {
-            return createTaskError(TaskConstants.SCHEDULER_ERROR, e.getMessage());
+            taskScheduler.resume(triggerID);
+        } catch (SchedulerException e) {
+            return Utils.createTaskError(e.getMessage());
         }
         return null;
     }
 
     public static Object detach(BObject taskListener, BObject service) {
+        TaskScheduler taskScheduler = (TaskScheduler) taskListener.getNativeData(TaskConstants.SCHEDULER);
         try {
-            Task task = (Task) taskListener.getNativeData(NATIVE_DATA_TASK_OBJECT);
-            String serviceName = service.getType().getName();
-            task.removeService(serviceName);
-        } catch (Exception e) {
-            return createTaskError(TaskConstants.SCHEDULER_ERROR, e.getMessage());
+            taskScheduler.removeService(service);
+        } catch (SchedulerException e) {
+            return Utils.createTaskError(e.getMessage());
         }
         return null;
     }
 
     public static Object start(BObject taskListener) {
-        Task task = (Task) taskListener.getNativeData(NATIVE_DATA_TASK_OBJECT);
-        TaskServerConnector serverConnector = new TaskServerConnectorImpl(task);
+        TaskScheduler taskScheduler = (TaskScheduler) taskListener.getNativeData(TaskConstants.SCHEDULER);
         try {
-            serverConnector.start();
-        } catch (SchedulingException e) {
-            return createTaskError(e.getMessage());
+            taskScheduler.start();
+        } catch (SchedulerException e) {
+            return Utils.createTaskError(e.getMessage());
         }
         return null;
     }
 
     public static Object stop(BObject taskListener) {
-        Task task = (Task) taskListener.getNativeData(NATIVE_DATA_TASK_OBJECT);
-        TaskServerConnector serverConnector = new TaskServerConnectorImpl(task);
+        String triggerID = (String) taskListener.getNativeData(TaskConstants.TRIGGER_NAME);
+        TaskScheduler taskScheduler = (TaskScheduler) taskListener.getNativeData(TaskConstants.SCHEDULER);
         try {
-            serverConnector.stop();
-        } catch (SchedulingException e) {
-            return createTaskError(e.getMessage());
+            taskScheduler.stop(triggerID);
+        } catch (SchedulerException e) {
+            return Utils.createTaskError(e.getMessage());
         }
         return null;
     }
 
+    @SuppressWarnings("unchecked")
```

Comment:
Shall we put an explanatory comment as to why this unchecked warning is being suppressed?